dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/EventRequestParamsMapper.java

```diff
@@ -231,23 +236,39 @@ private void validateUpdateDurationParams(RequestParams requestParams)
     }
   }
 
-  private void validateOrgUnitMode(RequestParams params) throws BadRequestException {
-    if ((params.getOrgUnitMode().equals(ACCESSIBLE) || params.getOrgUnitMode().equals(CAPTURE))
-        && params.getOrgUnit() != null) {
+  private void validateOrgUnitMode(UID orgUnit, OrganisationUnitSelectionMode orgUnitMode)
+      throws BadRequestException {
+    if ((orgUnitMode.equals(ACCESSIBLE) || orgUnitMode.equals(CAPTURE)) && orgUnit != null) {
```

Comment:
We should compare enums using `==` instead of `.equals` as `==` is null-safe. Even if you already checked that the mode cannot be null its a safer way to compare enums in general.